deploy/cluster_roles.yaml

```diff
@@ -0,0 +1,59 @@
+# this file contains the clusterroles needed for the core-functionality of the
+# operator.
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: osd-metrics-exporter-watch-clusterscope
+rules:
+  - apiGroups:
+      - config.openshift.io
+    resources:
+      - oauths
+    verbs:
+      - get
+      - list
+      - watch
+      - update
+  - apiGroups:
+      - config.openshift.io
+    resources:
+      - proxies
+      - clusterversions
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - rbac.authorization.k8s.io
+    resources:
+      - clusterroles
+    verbs:
+      - get
+      - list
+      - watch
+      - update
+  - apiGroups:
+      - user.openshift.io
+    resources:
+      - groups
+    verbs:
+      - get
+      - list
+      - watch
+      - update
+---
+# This clusterrole should only be bound in the openshift-osd-metrics and
+# openshift-config namespaces and not globally
+kind: ClusterRole
```

Comment:
I think anything in this file will get an automatic ClusterRoleBinding by OLM, so it may be better to roll this out into two seperate role/rolebinding pairs applied by a SyncSet. See cloud-ingress-operator in the `resources` folder for examples